lighthouse-core/report/html/renderer/element-screenshot-renderer.js

```diff
@@ -0,0 +1,237 @@
+          clipRect,
+          fullPageScreenshot,
+          {
+            // TODO: should this be documentElement width?
+            width: window.innerWidth * 0.75,
+            height: window.innerHeight * 0.75,
+          }
+        ));
+        document.body.appendChild(overlay);
+        overlay.addEventListener('click', () => {
+          overlay.remove();
+        });
+      });
+    }
+  }
+
+  /**
+   * @param {DOM} dom
+   * @param {ParentNode} templateContext
+   * @param {LH.Artifacts.Rect} clipRect Region of screenshot to highlight.
+   * @param {LH.Artifacts.FullPageScreenshot} fullPageScreenshot
+   * @param {Size} viewportSize Size of region to render screenshot in.
+   * @return {Element}
+   */
+  static render(dom, templateContext, clipRect, fullPageScreenshot, viewportSize) {
+    const tmpl = dom.cloneTemplate('#tmpl-lh-element-screenshot', templateContext);
+    const containerEl = dom.find('.lh-element-screenshot', tmpl);
+
+    let zoomFactor = 1;
+    const viewport = {
+      width: viewportSize.width,
+      height: viewportSize.height,
+    };
+
+    containerEl.setAttribute('rectWidth', clipRect.width.toString());
+    containerEl.setAttribute('rectHeight', clipRect.height.toString());
+    containerEl.setAttribute('rectLeft', clipRect.left.toString());
+    containerEl.setAttribute('rectRight', clipRect.right.toString());
+    containerEl.setAttribute('rectTop', clipRect.top.toString());
+    containerEl.setAttribute('rectBottom', clipRect.bottom.toString());
+
+    // Zoom out (by double viewport but using 0.5 zoom factor) when highlighted region takes
+    // up most of the viewport. This provides more context for where on the page this element is.
+    /* todo: maybe only apply the width criterium in the preview screenshot (what's in the table, not the overlay) */
+    // if (clipRect.height > viewportSize.height / 2 || clipRect.width > viewportSize.width / 2) {
+    //   zoomFactor = 0.5;
+    //   viewport.width *= 2;
+    //   viewport.height *= 2;
+    // }
+
+    const intendedClipToViewportRatio = 0.75;
+    const zoomRatioXY = {
+      x: viewportSize.width / clipRect.width,
+      // x: 1, // ignore x?
+      y: viewportSize.height / clipRect.height,
+    };
+    const zoomRatio = intendedClipToViewportRatio * Math.min(zoomRatioXY.x, zoomRatioXY.y);
+    if (zoomRatio < 1) {
+      zoomFactor = zoomRatio;
+      viewport.width /= zoomRatio;
+      viewport.height /= zoomRatio;
+    }
+
+    viewport.width = Math.min(fullPageScreenshot.width, viewport.width);
+
+    const positions = ElementScreenshotRenderer.getScreenshotPositions(
+      clipRect,
+      viewport,
+      {width: fullPageScreenshot.width, height: fullPageScreenshot.height}
+    );
+
+    const contentEl = dom.find('.lh-element-screenshot__content', containerEl);
+    // TODO: can all the `* zoomFactor` be replaces with setting some CSS on the container?
+    // just `scale` doesn't work b/c won't change size of the element.
```

Comment:
I'm not sure I understand this comment. `scale` on the container definitely won't work but scale and position on the  content element should be workable

Reply:
Not able to crack this.